spec.md

```diff
@@ -83,28 +113,55 @@ $\operatorname{SPLIT}_C \in V_8 \rightarrow V_v$
 
 The configuration must satisfy $S_{\text{max}} \ge S_{\text{min}} > 0$.
 
+<!---
+NOTE
+
+It might help the clarity of what follows
+if we make parameterization by C implicit
+rather than repeating C in subscripts everywhere.
+-->
+
 ## Definitions
 
 We define the constant $W$, which we call the "window size," to be 64.
 
-The "split index" $I(X)$ of a sequence $X$ is either the smallest
-non-negative integer $i$ satisfying:
+<!---
+NOTE
+
+Fixing W at 64
+arbitrarily rules out using this section
+to describe and reason about hashsplit algorithms
+that it otherwise could.
 
-- $i \le |X|$ and
-- $S_{\text{max}} \ge i \ge S_{\text{min}}$ and
-- $H(\langle X_{i-W}, \dots, X_{i-1} \rangle) \mod 2^T = 0$
+I think fixing it at 64 is more properly a part of the recommendation section.
+-->
 
-...or $\operatorname{min}(|X|, S_{\text{max}})$, if no such $i$ exists. For the
-purposes of this definition we set $X_i = 0$ for $i < 0$.
+We define the predicate $q_C(X)$
+on a non-empty byte sequence $X$
+with respect to a configuration $C$
+to be:
 
-The “prefix” $P(X)$ of a non-empty sequence $X$ is $\langle X_0, \dots, X_{I(X)-1} \rangle$.
+- $\text{true}$ if $|X| = S_{\text{max}}$; otherwise
+- $\text{true}$ if $|X| \ge S_{\text{min}}$ and $H(\langle X_{\max(0,|X|-W)}, \dots, X_{|X|-1} \rangle) \mod 2^T = 0$
+  (i.e., the last $W$ bytes of $X$ hash to a value with at least $T$ trailing zeroes);
+  otherwise
+- $\text{false}$.
 
-The “remainder” $R(X)$ of a non-empty sequence $X$ is $\langle X_{I(X)}, \dots, X_{|X|-1} \rangle$.
+<!---
+NOTE
+
+This previously used H(<X_(|X|-W) ... X_(|X|-1)>) and defined X_i = 0 for i<0.
+However, this unnecessarily constrains the choice of hash function.
+If the hash function wants to treat input shorter than W as being prefixed by zeroes,
+it can specify that;
+but if it wants to handle input shorter than W differently,
+it should be allowed to do that too.
```

Comment:
This gets a little weird because the way the hash functions are currently specified is over a variable length buffer; it's actually well defined what RRS1 and CP32 are when `|X| != W`, but a rolling implementation won't be amenable to computing that...

I'm still somewhat inclined to just insist that `W <= S_min`, as it would obviate this entirely, and it doesn't really affect the obvious implementation, but @cole-miller wanted this defined so he could do something with iterators of hashes...

@cole-miller, how strongly do you feel about this? I'm still not sure I grok your use case...

Reply:
Yeah, I think I understand better now how `S_min >= W` solves this kind of problem. And since it really applies specifically to the splitting procedure, not to the hash computation step, the implementation strategy I had in mind isn't affected. So I'm fine with re-imposing the restriction.